Simplenote/Information/NoteInformationController.swift

```diff
@@ -53,23 +95,37 @@ final class NoteInformationController {
 private extension NoteInformationController {
     func startListeningForChanges() {
         noteChangesObserver.delegate = self
+
+        referencesController?.onDidChangeContent = { [weak self] _, _ in
+            self?.sendNewDataToObserver()
+        }
     }
 
     func stopListeningForChanges() {
         noteChangesObserver.delegate = nil
+        referencesController?.onDidChangeContent = nil
     }
 }
 
 // MARK: - Data
 //
 private extension NoteInformationController {
-    func allRows() -> [Row] {
-        return metricRows()
+    func allSections() -> [Section] {
+        var sections: [Section] = []
+        sections.append(metricSection())
+        if let referenceSection = referenceSection() {
+            let header = Section(rows: [
+                Row.header(title: Localization.references.localizedUppercase)
+            ])
+            sections.append(header)
+            sections.append(referenceSection)
+        }
+        return sections
```

Comment:
Nipticky: Since the `referenceSection` comes with the Header as well (and both are part of the same thing!) would it be worth it renaming `referenceSection` > `referenceSections` and having such API return the header as well?

(That way this API would simplify!)

Reply:
Sounds good, thanks!